### PR TITLE
TEST-#3364: add value_counts benchmark for OmniSci backend (dataframe case)

### LINKED ISSUE #3364: ASV: add value_counts benchmark for OmniSci backend (dataframe case)

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #3364 
- [x] tests added and passing
- [x] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

File: asv_bench/benchmarks/omnisci/benchmarks.py [hunk 1]
```diff
@@ -256,25 +256,43 @@ def create_fillna_value(value_type: str, columns: list):
...
-class TimeValueCountsSeries:
-    param_names = ["shape", "ngroups"]
-    params = [
-        SERIES_DATA_SIZE[ASV_DATASET_SIZE],
-        GROUPBY_NGROUPS[ASV_DATASET_SIZE],
-    ]
-
-    def setup(self, shape, ngroups):
+class BaseTimeValueCounts:
+    def setup(self, shape, ngroups=5, subset=1):
```
Comment (dchigarev) [inline asv_bench/benchmarks/omnisci/benchmarks.py:260]: If the only change from modin on ray benchmarks is the data import triggering, could we just inherit from the modin-on-ray benchmarks without rewriting them all?

```python
from asv_bench.benchmarks.benchmarks import (
    BaseTimeValueCounts,
    TimeValueCountsSeries,
    TimeValueCountsFrame,
)

class BaseTimeValueCountsOmnisci(BaseTimeValueCounts):
    def setup(*args, **kwargs):
        super().setup(*args, **kwargs)
        trigger_import(self.df)

class TimeValueCountsSeriesOmnisci(BaseTimeValueCountsOmnisci, TimeValueCountsSeries):
    pass

class TimeValueCountsFrameOmnisci(BaseTimeValueCountsOmnisci, TimeValueCountsFrame):
    pass
```

Comment (anmyachev) [reply asv_bench/benchmarks/omnisci/benchmarks.py:260]: For series, we doesn't benchmark `bins` parameter.

In addition, benchmarks differ in data size.
For ray - `UNARY_OP_DATA_SIZE` also include `(5000, 5000)` shape.

**Comment by anmyachev** [comment]
@dchigarev @YarShev ready for review

**Comment by YarShev** [comment]
@anmyachev , job with flake8 failed, take a look.

**Comment by YarShev** [comment]
@anmyachev , something went wrong with asv job.

**Comment by anmyachev** [comment]
> @anmyachev , something went wrong with asv job.

yeh, something wrong with `skiprows`. @amyskov is looking at this.

**Comment by anmyachev** [comment]
blocked by https://github.com/modin-project/modin/pull/3391

**Comment by anmyachev** [comment]
@YarShev @dchigarev ready for review

**Review by dchigarev — APPROVED** [review]
LGTM

**Review by YarShev — APPROVED** [review]

File: asv_bench/benchmarks/omnisci/benchmarks.py [hunk 2]
```diff
@@ -244,25 +244,43 @@ def create_fillna_value(value_type: str, columns: list):
...
             RAND_HIGH,
-            groupby_ncols=1,
+            groupby_ncols=subset,
             count_groups=ngroups,
         )
-        self.series = self.df[self.column_names[0]]
+        trigger_import(self.df)
+
+
+class TimeValueCountsDataFrame(BaseTimeValueCounts):
```
Comment (YarShev) [inline asv_bench/benchmarks/omnisci/benchmarks.py:262]: The following change should be introduced as part of a separate issue for pandas on ray benchmarks. `TimeValueCountsFrame` -> `TimeValueCountsDataFrame`

**Pull request merged**